authorization/token/service/token_service.go

```diff
@@ -233,15 +233,27 @@ func (s *tokenServiceImpl) Audit(ctx context.Context, identity *account.Identity
 
 				oldPrivResourceID := oldPriv.ResourceID
 
-				// Create a new permissions object for the RPT token and store it in the array
-				perm := &token.Permissions{
-					ResourceSetID: &oldPrivResourceID,
-					Scopes:        oldPriv.ScopesAsArray(),
-					Expiry:        oldPriv.ExpiryTime.Unix(),
+				// If the old privilege is stale, then refresh its scopes and expiry time
+				if oldPriv.Stale {
+					// Lookup the scopes for the old privilege, as they may have changed
+					privilegeCache, err = s.Services().PrivilegeCacheService().CachedPrivileges(ctx, identity.ID, oldPrivResourceID)
+					if err != nil {
+						return errors.NewInternalError(ctx, err)
```

Comment:
unless the underlying error should explicitly become an `Internal Error`, we should just `wrap` it with a message about the context of the call

Reply:
Good catch, the underlying error here will already be an `InternalError` (as that is all the `CachedPrivileges()` function can return) so no need to wrap it again.